controllers/k8ssandra/medusa_reconciler.go

```diff
@@ -0,0 +1,239 @@
+package k8ssandra
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	api "github.com/k8ssandra/k8ssandra-operator/apis/k8ssandra/v1alpha1"
+	medusaapi "github.com/k8ssandra/k8ssandra-operator/apis/medusa/v1alpha1"
+	cassandra "github.com/k8ssandra/k8ssandra-operator/pkg/cassandra"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/secret"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Generate a secret for Medusa or use the existing one if provided in the spec
+func (r *K8ssandraClusterReconciler) reconcileMedusaSecrets(
+	ctx context.Context,
+	kc *api.K8ssandraCluster,
+	logger logr.Logger,
+) error {
+	logger.Info("Reconciling Medusa user secrets")
+	if kc.Spec.Medusa != nil {
+		cassandraUserSecretRef := kc.Spec.Medusa.CassandraUserSecretRef
+		if cassandraUserSecretRef == "" {
+			cassandraUserSecretRef = cassandraUserSecretName(kc.Spec.Medusa, kc.Name)
+		}
+		logger = logger.WithValues(
+			"MedusaCassandraUserSecretRef",
+			cassandraUserSecretRef,
+		)
+		if err := secret.ReconcileSecret(ctx, r.Client, cassandraUserSecretRef, kc.Name, kc.Namespace); err != nil {
+			logger.Error(err, "Failed to reconcile Medusa CQL user secret")
+			return err
+		}
+	}
+	logger.Info("Medusa user secrets successfully reconciled")
+	return nil
+}
+
+// Get the cassandra user secret name from the medusa spec or generate one based on the cluster name
+func cassandraUserSecretName(medusaSpec *medusaapi.MedusaClusterTemplate, clusterName string) string {
```

Comment:
Nit. The function name is misleading, it's about Medusa but the name says "cassandra"?